### PR TITLE
GetAccountState

### PR DESCRIPTION
**Pull request by Ashuaidehao**

### CODE REVIEW

**Review by shargon — APPROVED** [review]

**Comment by cloud8little** [comment]
Test passed:

Account A vote to candidate a with 2 vote, and then transfer 1 neo out, all scenarios work.

```
{
  "jsonrpc": "2.0",
  "method": "invokefunction",
  "params": ["0xef4073a0f2b305a38ec4050e4d3d28bc40ea63f5","getAccountState",[{"type":"Hash160","value":"0x2186ac696d435bdb75c8761373473936dd645beb"}											],[
	]
],
  "id": 1
}
```

```
{
  "jsonrpc": "2.0",
  "id": 1,
  "result": {
    "script": "DBTrW2TdNjlHcxN2yHXbW0NtaayGIRHAHwwPZ2V0QWNjb3VudFN0YXRlDBT1Y+pAvCg9TQ4FxI6jBbPyoHNA70FifVtS",
    "state": "HALT",
    "gasconsumed": "2028330",
    "exception": null,
    "stack": [
      {
        "type": "Struct",
        "value": [
          {
            "type": "Integer",
            "value": "1"
          },
          {
            "type": "Integer",
            "value": "252"
          },
          {
            "type": "ByteString",
            "value": "AwgyqZUbsv67SMlHvO2NsnmBiF8l4lBpp139EkH2q71z"
          }
        ]
      }
    ]
  }
}
```

the first item is current balance, 2nd item is update height, 3rd item is the voteTo pubkey.

**Review by cloud8little — DISMISSED** [review]

File: src/neo/SmartContract/Native/NeoToken.cs [hunk 1]
```diff
@@ -348,6 +348,18 @@ public ECPoint[] GetCommittee(DataCache snapshot)
             return GetCommitteeFromCache(snapshot).Select(p => p.PublicKey).OrderBy(p => p).ToArray();
         }
 
+        /// <summary>
+        /// Get account state.
+        /// </summary>
+        /// <param name="snapshot">The snapshot used to read data.</param>
+        /// <param name="account">account</param>
+        /// <returns>The amount of the fees.</returns>
```
Comment (erikzhang) [inline src/neo/SmartContract/Native/NeoToken.cs:356]: Can you update the comments?

File: src/neo/SmartContract/Native/NeoToken.cs [hunk 2]
```diff
@@ -348,6 +348,18 @@ public ECPoint[] GetCommittee(DataCache snapshot)
...
+        [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.ReadStates)]
+        public NeoAccountState GetAccountState(DataCache snapshot, UInt160 account)
+        {
+            return snapshot.GetAndChange(CreateStorageKey(Prefix_Account).Add(account))?.GetInteroperable<NeoAccountState>();
```
Comment (erikzhang) [inline src/neo/SmartContract/Native/NeoToken.cs:360]: You should use `TryGet` instead of `GetAndChange`.

**Review by erikzhang — CHANGES_REQUESTED** [review]

File: src/neo/SmartContract/Native/NeoToken.cs (hunk 2 above)
Comment (shargon) [reply src/neo/SmartContract/Native/NeoToken.cs]: Done

File: src/neo/SmartContract/Native/NeoToken.cs (hunk 1 above)
Comment (shargon) [reply src/neo/SmartContract/Native/NeoToken.cs]: Done

**Review by shargon — APPROVED** [review]

**Review by erikzhang — APPROVED** [review]

**Pull request merged**